### DISCUSSION TITLE
The problem of fitting Al/SiC interface model

### DISCUSSION THREAD

**Discussion by linkx-stu** [deepmodeling/deepmd-kit#2706] [Q&A]
hello!
          I prepared three types of AIMD data: 1. bulk-Al, 2. bulk-SiC, 3. interface-Al/SiC.
          Therefore, the number of atomic types in our dataset may not be consistent. In the SiC dataset, the C atom is labeled as 0, and the Si atom is labeled as 1. In the Al dataset, the Al atom is labeled as 0. In the Al/SiC interface dataset, the aluminum atom is labeled as 0, the carbon atom is labeled as 1, and the silicon atom is labeled as 2.
         How to ensure that the order of element types in the **type_map** parameter is consistent with the numbering of atomic types in the dataset?

**Comment by njzjz** [comment]
Please see [documentation](https://docs.deepmodeling.com/projects/deepmd/en/master/data/system.html)

> Integers that start with 0. If both the training parameter [type_map](https://docs.deepmodeling.com/projects/deepmd/en/master/train/train-input.html#model-type-map) is set and type_map.raw is provided, the system atom type should be mapped to type_map.raw in type.raw and will be mapped to the model atom type when training; otherwise, the system atom type will be always mapped to the model atom type (whether [type_map](https://docs.deepmodeling.com/projects/deepmd/en/master/train/train-input.html#model-type-map) is set or not)

**Reply by linkx-stu** [reply]
Part of my setup is as follows：

type_map is set as:
"type_map": [
"Al",
"C",
"Si"
],

type_map.raw is provided:
Al C Si
My Bulk-SiC training dataset contains only C and Si atoms，so type.raw shows as:
1 2
My Bulk-Al training dataset contains only Al atoms，so type.raw shows as:
0

 "training": {
    "training_data": {
      "systems": [
        "../data/data-sic",
        "../data/data-al"
      ],
      "auto_prob": "prob_sys_size; 0:1:0.5; 1:2:0.5",
      "batch_size": "auto",
      "_comment": "that's all"
    },
    "validation_data": {
      "systems": [
        "../data/validation_data"
         
    Is this parameter setting reasonable?

**Reply by njzjz** [reply]
It looks ok.

**Reply by StealFox97** [reply]
@linkuixin  Can you tell me how you changed the contents of type_map.raw and type.raw files? Can it be changed directly through the 'vim' command? Is it OK?

**Reply by linkx-stu** [reply]
dpdata can do this, such as：
import dpdata
data = dpdata.LabeledSystem('OUTCAR')
type_map = ['element-1', 'element-2', 'element-3']
data.apply_type_map(type_map)
data.to('deepmd/npy', 'data', set_size=400)

**Reply by ZKNB** [reply]
If you don't mind, can you provide your input.json file for reference, I'm having this problem too.
[https://github.com/deepmodeling/deepmd-kit/discussions/2988](url)
